api/C/include/csi_ll_pmp.h

```diff
@@ -0,0 +1,237 @@
+/*
+ * Low-Level PMP API
+ *
+ * This module provides functionality for PMP configuration.
+ *
+ * Copyright (C) 2017-2023 Alibaba Group Holding Limited
+ */
+
+#ifndef CSI_LL_PMP_H
+#define CSI_LL_PMP_H
+
+#include "csi_types.h"
+#include <stdint.h>
+
+/*
+ * Types of address Matching
+ */
+typedef enum {
+    ADDRESS_MATCHING_OFF   = 0,	/* Null region (disabled) */
+    ADDRESS_MATCHING_TOR   = 1, /* Top of range */
+    ADDRESS_MATCHING_NA4   = 2, /* Naturally aligned four-byte region */
+    ADDRESS_MATCHING_NAPOT = 3, /* Naturally aligned power-of-two region, ≥8 bytes */
+} csi_addr_matching_t;
+
+/*
+ * Attribution of PMP coniguration
+ */
+typedef struct {
+    bool flag_r; /* The flag of readable. 0: not readable; 1: readable */
+    bool flag_w; /* The flag of writable. 0: not writable; 1: writable */
+    bool flag_x; /* The flag of execable. 0: not execable; 1: execable */
+    csi_addr_matching_t mode; /* Address matching mode */
+    bool lock; /* The flag of lock,  0: not be locked; 1 : be locked */
+} csi_pmp_region_attr_t;
+
+/*
+ * Get PMP range number for the hard id.
+ *
+ * If the hard id is invalidate, such as out of support range, it should return
+ * an error
+ *
+ * @param hart_id: The hart id used to get the pmp range number
+ * @param range_number: A pointer to store the ouptut value of PMP range number
+ * @return : Status code
+ */
+csi_status_t csi_pmp_get_range_num(unsigned int hart_id, unsigned int *range_number);
+
+/*
+ * Configure PMP entry.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
```

Comment:
I think only one result when called in other mode should be defined, such as it is not permit call it in other mode except m, it will just crash.